WordPress/MediaEditor/MediaEditorHub.swift

```diff
@@ -78,16 +117,128 @@ class MediaEditorHub: UIViewController {
         activityIndicatorView.isHidden = true
     }
 
+    private func reloadImagesAndReposition() {
+        thumbsCollectionView.reloadData()
+        imagesCollectionView.reloadData()
+        thumbsCollectionView.layoutIfNeeded()
+        thumbsCollectionView.selectItem(at: IndexPath(row: 0, section: 0), animated: false, scrollPosition: .left)
+        imagesCollectionView.scrollToItem(at: IndexPath(row: self.selectedThumbIndex, section: 0), at: .right, animated: false)
+        thumbsToolbar.isHidden = numberOfThumbs > 1 ? false : true
+    }
+
     private func setupForOrientation() {
         let isLandscape = UIDevice.current.orientation.isLandscape
-        portraitConstraints.forEach { $0.isActive = !isLandscape }
-        landscapeConstraints.forEach { $0.isActive = isLandscape }
+        mainStackView.axis = isLandscape ? .horizontal : .vertical
+        mainStackView.semanticContentAttribute = isLandscape ? .forceRightToLeft : .unspecified
         horizontalToolbar.isHidden = isLandscape
         verticalToolbar.isHidden = !isLandscape
+        if let layout = thumbsCollectionView.collectionViewLayout as? UICollectionViewFlowLayout {
+            layout.scrollDirection = isLandscape ? .vertical : .horizontal
+        }
+        mainStackView.layoutIfNeeded()
+        imagesCollectionView.scrollToItem(at: IndexPath(row: selectedThumbIndex, section: 0), at: .right, animated: false)
+    }
+
+    private func highlightSelectedThumb(current: Int, before: Int) {
+        let current = thumbsCollectionView.cellForItem(at: IndexPath(row: current, section: 0)) as? MediaEditorThumbCell
+        let before = thumbsCollectionView.cellForItem(at: IndexPath(row: before, section: 0)) as? MediaEditorThumbCell
+        before?.hideBorder()
+        current?.showBorder()
+    }
+
+    private func showOrHideActivityIndicator() {
+        let imageAvailable = availableThumbs[selectedThumbIndex] ?? availableImages[selectedThumbIndex]
+
+        imageAvailable == nil ? showActivityIndicator() : hideActivityIndicator()
     }
 
     static func initialize() -> MediaEditorHub {
         return UIStoryboard(name: "MediaEditorHub", bundle: nil).instantiateViewController(withIdentifier: "hubViewController") as! MediaEditorHub
     }
 
+    private enum Constants {
+        static var thumbCellIdentifier = "thumbCell"
+        static var imageCellIdentifier = "imageCell"
+    }
+}
+
+// MARK: - UICollectionViewDataSource
+
+extension MediaEditorHub: UICollectionViewDataSource {
+    func collectionView(_ collectionView: UICollectionView, numberOfItemsInSection section: Int) -> Int {
+        return numberOfThumbs
+    }
+
+    func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {
+        // Thumbs Collection View
```

Comment:
Suggestion: Whenever I see comments like this one and the one below (`// Images Collection View`), I tend to believe code can be separated even further, in order to simplify things and improve readability.

Something like:

```swift
func collectionView(_ collectionView: UICollectionView, cellForItemAt indexPath: IndexPath) -> UICollectionViewCell {

    if collectionView == thumbsCollectionView {
        return cellForThumbsCollectionView(...)
    } else {
        return cellForImagesCollectionView(...)
    }
}
```

Reply:
Done in 59c5776e7bd784c6e543e575f2e261dabb388217